In [1]:
import seaborn as sns
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import warnings
warnings.filterwarnings('ignore')

## Load the datasets contained in the seaborn library
### we can get names of the datasets in the library 

In [2]:
sns.get_dataset_names()

['anagrams',
 'anscombe',
 'attention',
 'brain_networks',
 'car_crashes',
 'diamonds',
 'dots',
 'dowjones',
 'exercise',
 'flights',
 'fmri',
 'geyser',
 'glue',
 'healthexp',
 'iris',
 'mpg',
 'penguins',
 'planets',
 'seaice',
 'taxis',
 'tips',
 'titanic']

In [3]:
df=sns.load_dataset('iris')
df

sepal_length  sepal_width  petal_length  petal_width    species
0             5.1          3.5           1.4          0.2     setosa
1             4.9          3.0           1.4          0.2     setosa
2             4.7          3.2           1.3          0.2     setosa
3             4.6          3.1           1.5          0.2     setosa
4             5.0          3.6           1.4          0.2     setosa
..            ...          ...           ...          ...        ...
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           6.2          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica

[150 rows x 5 columns]

In [15]:
df1=sns.load_dataset('flights')
df1

year month  passengers
0    1949   Jan         112
1    1949   Feb         118
2    1949   Mar         132
3    1949   Apr         129
4    1949   May         121
5    1949   Jun         135
6    1949   Jul         148
7    1949   Aug         148
8    1949   Sep         136
9    1949   Oct         119
10   1949   Nov         104
11   1949   Dec         118
12   1950   Jan         115
13   1950   Feb         126
14   1950   Mar         141
15   1950   Apr         135
16   1950   May         125
17   1950   Jun         149
18   1950   Jul         170
19   1950   Aug         170
20   1950   Sep         158
21   1950   Oct         133
22   1950   Nov         114
23   1950   Dec         140
24   1951   Jan         145
25   1951   Feb         150
26   1951   Mar         178
27   1951   Apr         163
28   1951   May         172
29   1951   Jun         178
30   1951   Jul         199
31   1951   Aug         199
32   1951   Sep         184
33   1951   Oct         162
34   1951   Nov         146
35   1951   Dec         166
36   1952   Jan         171
37   1952   Feb         180
38   1952   Mar         193
39   1952   Apr         181
40   1952   May         183
41   1952   Jun         218
42   1952   Jul         230
43   1952   Aug         242
44   1952   Sep         209
45   1952   Oct         191
46   1952   Nov         172
47   1952   Dec         194
48   1953   Jan         196
49   1953   Feb         196
50   1953   Mar         236
51   1953   Apr         235
52   1953   May         229
53   1953   Jun         243
54   1953   Jul         264
55   1953   Aug         272
56   1953   Sep         237
57   1953   Oct         211
58   1953   Nov         180
59   1953   Dec         201
60   1954   Jan         204
61   1954   Feb         188
62   1954   Mar         235
63   1954   Apr         227
64   1954   May         234
65   1954   Jun         264
66   1954   Jul         302
67   1954   Aug         293
68   1954   Sep         259
69   1954   Oct         229
70   1954   Nov         203
71   1954   Dec         229
72   1955   Jan         242
73   1955   Feb         233
74   1955   Mar         267
75   1955   Apr         269
76   1955   May         270
77   1955   Jun         315
78   1955   Jul         364
79   1955   Aug         347
80   1955   Sep         312
81   1955   Oct         274
82   1955   Nov         237
83   1955   Dec         278
84   1956   Jan         284
85   1956   Feb         277
86   1956   Mar         317
87   1956   Apr         313
88   1956   May         318
89   1956   Jun         374
90   1956   Jul         413
91   1956   Aug         405
92   1956   Sep         355
93   1956   Oct         306
94   1956   Nov         271
95   1956   Dec         306
96   1957   Jan         315
97   1957   Feb         301
98   1957   Mar         356
99   1957   Apr         348
100  1957   May         355
101  1957   Jun         422
102  1957   Jul         465
103  1957   Aug         467
104  1957   Sep         404
105  1957   Oct         347
106  1957   Nov         305
107  1957   Dec         336
108  1958   Jan         340
109  1958   Feb         318
110  1958   Mar         362
111  1958   Apr         348
112  1958   May         363
113  1958   Jun         435
114  1958   Jul         491
115  1958   Aug         505
116  1958   Sep         404
117  1958   Oct         359
118  1958   Nov         310
119  1958   Dec         337
120  1959   Jan         360
121  1959   Feb         342
122  1959   Mar         406
123  1959   Apr         396
124  1959   May         420
125  1959   Jun         472
126  1959   Jul         548
127  1959   Aug         559
128  1959   Sep         463
129  1959   Oct         407
130  1959   Nov         362
131  1959   Dec         405
132  1960   Jan         417
133  1960   Feb         391
134  1960   Mar         419
135  1960   Apr         461
136  1960   May         472
137  1960   Jun         535
138  1960   Jul         622
139  1960   Aug         606
140  1960   Sep         508
141  1960   Oct         461
1

## Task:
### Load all the datasets and do the EDA

## 2. Scikit-learn datasets

In [5]:
from sklearn.datasets import fetch_california_housing

## Load data (will download the data if its the first time loading)

In [6]:
housing=fetch_california_housing(as_frame=True)

### Create  a dataframe


In [7]:
df=housing['data'].join(housing['target'])
df

MedInc  HouseAge  AveRooms  AveBedrms  Population  AveOccup  Latitude  \
0      8.3252      41.0  6.984127   1.023810       322.0  2.555556     37.88   
1      8.3014      21.0  6.238137   0.971880      2401.0  2.109842     37.86   
2      7.2574      52.0  8.288136   1.073446       496.0  2.802260     37.85   
3      5.6431      52.0  5.817352   1.073059       558.0  2.547945     37.85   
4      3.8462      52.0  6.281853   1.081081       565.0  2.181467     37.85   
...       ...       ...       ...        ...         ...       ...       ...   
20635  1.5603      25.0  5.045455   1.133333       845.0  2.560606     39.48   
20636  2.5568      18.0  6.114035   1.315789       356.0  3.122807     39.49   
20637  1.7000      17.0  5.205543   1.120092      1007.0  2.325635     39.43   
20638  1.8672      18.0  5.329513   1.171920       741.0  2.123209     39.43   
20639  2.3886      16.0  5.254717   1.162264      1387.0  2.616981     39.37   

       Longitude  MedHouseVal  
0        -122.23        4.526  
1        -122.22        3.585  
2        -122.24        3.521  
3        -122.25        3.413  
4        -122.25        3.422  
...          ...          ...  
20635    -121.09        0.781  
20636    -121.21        0.771  
20637    -121.22        0.923  
20638    -121.32        0.847  
20639    -121.24        0.894  

[20640 rows x 9 columns]

### Task : Load other datasets and do the EDA

## 3.pydataset

#### You may need to pip install the pydataset library

In [8]:
from pydataset import data

In [9]:
data()

dataset_id                                             title
0    AirPassengers       Monthly Airline Passenger Numbers 1949-1960
1          BJsales                 Sales Data with Leading Indicator
2              BOD                         Biochemical Oxygen Demand
3     Formaldehyde                     Determination of Formaldehyde
4     HairEyeColor         Hair and Eye Color of Statistics Students
..             ...                                               ...
752        VerbAgg                  Verbal Aggression item responses
753           cake                 Breakage Angle of Chocolate Cakes
754           cbpp                 Contagious bovine pleuropneumonia
755    grouseticks  Data on red grouse ticks from Elston et al. 2001
756     sleepstudy       Reaction times in a sleep deprivation study

[757 rows x 2 columns]

In [10]:
data().iloc[200]

dataset_id                  Florida
title         Florida County Voting
Name: 200, dtype: object

In [11]:
df2=data('Florida')
df2

GORE    BUSH  BUCHANAN  NADER  BROWNE  HAGELIN  HARRIS  \
ALACHUA        47300   34062       262   3215     658       42       4   
BAKER           2392    5610        73     53      17        3       0   
BAY            18850   38637       248    828     171       18       5   
BRADFORD        3072    5413        65     84      28        2       0   
BREVARD        97318  115185       570   4470     643       39      11   
BROWARD       386518  177279       789   7099    1212      128      49   
CALHOUN         2155    2873        90     39      10        1       0   
CHARLOTTE      29641   35419       182   1461     127       15       6   
CITRUS         25501   29744       270   1378     194       16       5   
CLAY           14630   41745       186    562     204       14       1   
COLLIER        29905   60426       122   1400     185       34       7   
COLUMBIA        7047   10964        89    258     127        7       1   
DADE          328702  289456       561   5355     759      119      88   
DE.SOTO         3322    4256        36    157      23        0       0   
DIXIE           1825    2698        29     75      32        2       0   
DUVAL         107680  152082       650   2752     954      160      36   
ESCAMBIA       40958   73029       504   1729     297       24       6   
FLAGLER        13891   12608        83    435      60        4       1   
FRANKLIN        2042    2448        33     85      17        3       1   
GADSDEN         9565    4750        39    139      24        4       7   
GILCHRIST       1910    3300        29     97      52        1       0   
GLADES          1420    1840         9     56      12        3       0   
GULF            2389    3546        71     86      21        4       2   
HAMILTON        1718    2153        24     39      11        2       6   
HARDEE          2341    3764        30     75      17        2       0   
HENDRY          3239    4743        22    103      11        1       3   
HERNANDO       32644   30646       242   1501     116       26       8   
HIGHLANDS      14152   20196        99    409      51       14       5   
HILLSBOROUGH  166581  176967       836   7348    1104      215      35   
HOLMES          2154    4985        76     91      18        7       1   
INDIAN.RIVER   19769   28627       105    950     123       13       4   
JACKSON         6868    9138       102    138      40        2       0   
JEFFERSON       3038    2481        29     76      14        1       2   
LAFAYETTE        788    1669        10     26       6        0       3   
LAKE           36555   49963       289   1459     203       36       4   
LEE            73560  106141       305   3587     538       81      30   
LEON           61425   39053       282   1932     330       28       9   
LEVY            5403    6860        67    284      92        1       1   
LIBERTY         1011    1316        39     19      12        3       0   
MADISON         3011    3038        29     54      18        2       0   
MANATEE        49169   57948       272   2489     243       35       5   
MARION         44648   55135       563   1810     361       26      14   
MARTIN         26619   33864       108   1075     105       29      13   
MONROE         16483   16059        47   1090     162       26       1   
NASSAU          6952   16404        90    255      63        8       0   
OKALOOSA       16924   52043       267    984     313       15       4   
OKEECHOBEE      4588    5058        43    131      21        4       1   
ORANGE        140115  134476       446   3881     892       65      13   
OSCEOLA        28177   26216       145    732     309       21      10   
PALM.BEACH    268945  152846      3407   5564     743      143      45   
PASCO          69550   68581       570   3392     412       82      18   
PINELLAS      199660  184312      1010   9986    1222      444      40   
POLK           74977   90101       538   2060     365       59       8   
PUTNAM         12091

In [12]:
data().iloc[1]

dataset_id                              BJsales
title         Sales Data with Leading Indicator
Name: 1, dtype: object

### These are the names of data sets available,There are currently 757 Dtasets.

## Loading the BOD dataset

In [13]:
df=data('BOD')
df

Time  demand
1     1     8.3
2     2    10.3
3     3    19.0
4     4    16.0
5     5    15.6
6     7    19.8

### Task:
    Load other datasets and do the EDA

In [14]:
df1=data('HairEyeColor')
df1

Hair    Eye     Sex  Freq
1   Black  Brown    Male    32
2   Brown  Brown    Male    53
3     Red  Brown    Male    10
4   Blond  Brown    Male     3
5   Black   Blue    Male    11
6   Brown   Blue    Male    50
7     Red   Blue    Male    10
8   Blond   Blue    Male    30
9   Black  Hazel    Male    10
10  Brown  Hazel    Male    25
11    Red  Hazel    Male     7
12  Blond  Hazel    Male     5
13  Black  Green    Male     3
14  Brown  Green    Male    15
15    Red  Green    Male     7
16  Blond  Green    Male     8
17  Black  Brown  Female    36
18  Brown  Brown  Female    66
19    Red  Brown  Female    16
20  Blond  Brown  Female     4
21  Black   Blue  Female     9
22  Brown   Blue  Female    34
23    Red   Blue  Female     7
24  Blond   Blue  Female    64
25  Black  Hazel  Female     5
26  Brown  Hazel  Female    29
27    Red  Hazel  Female     7
28  Blond  Hazel  Female     5
29  Black  Green  Female     2
30  Brown  Green  Female    14
31    Red  Green  Female     7
32  Blond  Green  Female     8